### PR TITLE
Micro-optimize icon smoothing subsystem, cutting down more than 50% of its init time (about .8s drop on local, more on prod)

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
With atoms now being 50% of the init time instead of ~80%, I've decided to poke around other subsystems.

Unfortunately they're not straight forward at all and extremely boring. `bitmask_smooth()` in particular is just a big proc that is called ~60k times, so I made some adjustments to its macro to reduce ops.

I removed code that looped over two entire Z-levels to find things to smooth, added by https://github.com/tgstation/tgstation/pull/16067/commits/e2dd404d39b0b9ac60d631a5476715c61ec6e37d, at a time before the initialize system. This doesn't have huge gains since it just defers to the queue anyway, but it still shaved off 0.1-0.2s on local.

I removed `update_appearance(~UPDATE_SMOOTHING)` from something since the person who added it doesn't remember why and it doesn't appear to do anything. Lemon thinks it's something to do with emissives, but I tested with neon carpet and it seemed to work fine.

### CODE REVIEW

File: code/__HELPERS/icon_smoothing.dm [hunk 1]
```diff
@@ -361,36 +361,46 @@ DEFINE_BITFIELD(smoothing_junction, list(
...
+			SOUTH_JUNCTION|EAST_JUNCTION|SOUTHEAST_JUNCTION,
+		)
+			icon_state = "[base_icon_state]-[smoothing_junction]-d"
+			if(new_junction == . || fixed_underlay) // Mutable underlays?
+				return .
+
+			var/junction_dir = reverse_ndir(smoothing_junction)
+			var/turned_adjacency = REVERSE_DIR(junction_dir)
+			var/turf/neighbor_turf = get_step(src, turned_adjacency & (NORTH|SOUTH))
+			var/mutable_appearance/underlay_appearance = mutable_appearance(layer = TURF_LAYER, plane = FLOOR_PLANE)
```
Comment (LemonInTheDark) [inline code/__HELPERS/icon_smoothing.dm:390]: It's significantly faster to static a mutable appearance, and then set variables on that. At least it was for atom movables.
I wonder if that would help at all here, or if this code just isn't very hot

Comment (Mothblocks) [reply code/__HELPERS/icon_smoothing.dm:390]: that's funny. i'll look

**Comment by Mothblocks** [comment]
Will look at this more some other time

**Comment by LemonInTheDark** [comment]
sadge

**Comment by Mothblocks** [comment]
i didnt need to close this

**Comment by Mothblocks** [comment]
but we should still test merge, its been a while

**Pull request merged**